domain/modelconfig/handlers/interface.go

```diff
@@ -0,0 +1,20 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package handlers
+
+import (
+	"context"
+)
+
+// ConfigHandler amends model config as it is being saved and loaded from state.
+type ConfigHandler interface {
```

Comment:
After reading a bit more of this PR I am worried that with handlers we could run into a situation of bad programming where people try to start doing cross cutting concerns in handlers that could violate transactional boundaries.

What I am particularly interested to see and talk about is the handler registering the set of keys that it cares about. It will only ever have the OnSave method called with those keys. The service will also do the job of removing those keys from model config as the handler has registered ownership.

Reply:
As discussed, handlers take set of attributes to operate on.